### PR TITLE
doc: add lockfile-path unstable doc section

### PR DESCRIPTION
**Pull request by Ifropc**
Adds lockfile-path into the unstable doc section as part of tracking issue https://github.com/rust-lang/cargo/issues/14421 tasks

### CODE REVIEW

File: src/doc/src/reference/unstable.md [hunk 1]
```diff
@@ -112,6 +112,8 @@ Each new feature described below should explain how to use it.
 * Registries
     * [publish-timeout](#publish-timeout) --- Controls the timeout between uploading the crate and being available in the index
     * [asymmetric-token](#asymmetric-token) --- Adds support for authentication tokens using asymmetric cryptography (`cargo:paseto` provider).
+* Mafinest and lockfile
```
Comment (Ifropc) [inline src/doc/src/reference/unstable.md:115]: Not sure if we should add another section, but existing ones don't seem to be relative enough for this feature, WDYT?

Comment (weihanglo) [reply src/doc/src/reference/unstable.md:115]: Maybe under Other suffices

**Review by weihanglo — APPROVED** [review]

**Comment by weihanglo** [comment]
Thank you for updating the doc!

@bors r+

**Pull request merged**